contracts/strategies/allocation/direct-grants/DirectGrantsStrategy.sol

```diff
@@ -0,0 +1,317 @@
+
+    bool public registryGating;
+    uint256 public upcomingMilestone;
+    address public acceptedRecipientId;
+
+    /// ===============================
+    /// ========== Events =============
+    /// ===============================
+
+    event Allocated(bytes data, address indexed allocator);
+    event MilestoneAccepted();
+    event MilestoneRejected(uint256 milestoneId);
+    event MilestoneSet();
+    event MilestoneSubmitted(uint256 milestoneId);
+    // Note: this event will emit 0x0000000000000000000000000000000000000000000000000000000000000000
+    // for identityId when not used.
+    event RecipientAdded(address indexed recipientId, Metadata metadata);
+    event RecipientStatusUpdated(address indexed recipientId, RecipientStatus status);
+
+    /// ===============================
+    /// ========= Modifiers ===========
+    /// ===============================
+
+    /// ===============================
+    /// ======== Constructor ==========
+    /// ===============================
+
+    constructor(address _allo, string memory _name) BaseStrategy(_allo, _name) {}
+
+    /// ===============================
+    /// ======== Initialize ===========
+    /// ===============================
+
+    // Note: expected _data shape:
+    // { bool registryGating } -> To be determined for the rest.
+    /// @notice Initialize the strategy
+    /// @param _poolId The pool ID for this strategy
+    /// @param _data The data to initialize the strategy with
+    function initialize(uint256 _poolId, bytes memory _data) public override {
+        super.initialize(_poolId, _data);
+
+        (registryGating) = abi.decode(_data, (bool));
+
+        registry = allo.getRegistry();
+    }
+
+    /// ===============================
+    /// =========== Views =============
+    /// ===============================
+
+    // Note: what determines a valid allocator?
+    function isValidAllocator(address _allocator) public view returns (bool) {
+        return allo.isPoolManager(poolId, _allocator);
+    }
+
+    function getRecipientStatus(address _recipientId) public view returns (RecipientStatus) {
+        return recipients[_recipientId].status;
+    }
+
+    /// @notice Get the status of the milestone
+    /// @param _milestoneId Id of the milestone
+    function getMilestoneStatus(uint256 _milestoneId) public view returns (RecipientStatus) {
+        // make sure it exists
+        if (milestones.length < _milestoneId) {
+            revert INVALID_MILESTONE();
+        }
+
+        return milestones[_milestoneId].milestoneStatus;
+    }
+
+    /// @notice Get the payout summary data for recipient
+    /// @param _recipientIds The recipient IDs to get the payout summary for. Note: this will only ever be 1 recipient for this strategy.
+    /// @param _data The data to get the payout summary for
+    /// @param _sender The sender of the transaction
+    function getPayouts(address[] memory _recipientIds, bytes memory _data, address _sender)
```

Comment:
We wouldn't need this right ?

Reply:
Agreed, the milestones will show the amount to payout and will control that.